project-base/storefront/components/Pages/Order/TransportAndPayment/helpers.ts

```diff
@@ -0,0 +1,356 @@
+import { ChangePaymentHandler } from 'hooks/cart/useChangePaymentInCart';
+import { ChangeTransportHandler } from 'hooks/cart/useChangeTransportInCart';
+import { useDomainConfig } from 'hooks/useDomainConfig';
+import getConfig from 'next/config';
+import { useEffect, useState } from 'react';
+import { usePersistStore } from 'store/usePersistStore';
+import { getGtmPickupPlaceFromStore, getGtmPickupPlaceFromLastOrder } from 'gtm/helpers/mappers';
+import { Translate } from 'next-translate';
+import { getInternationalizedStaticUrls } from 'helpers/getInternationalizedStaticUrls';
+import { useRouter } from 'next/router';
+import { useCurrentCustomerData } from 'connectors/customer/CurrentCustomer';
+import { useClient } from 'urql';
+
+const { publicRuntimeConfig } = getConfig();
+
+export const usePaymentChangeInSelect = (changePaymentHandler: ChangePaymentHandler) => {
+    const { payment: currentPayment, paymentGoPayBankSwift: currentPaymentGoPayBankSwift } = useCurrentCart();
+
+    const changePayment = async (updatedPaymentUuid: string | null) =>
+        await changePaymentHandler(updatedPaymentUuid, currentPaymentGoPayBankSwift);
+
+    const changeGoPaySwift = async (newGoPaySwiftValue: string | null) => {
+        await changePaymentHandler(currentPayment?.uuid ?? null, newGoPaySwiftValue);
+    };
+
+    const resetPaymentAndGoPayBankSwift = async () => await changePaymentHandler(null, null);
+
+    return { changePayment, changeGoPaySwift, resetPaymentAndGoPayBankSwift };
+};
+
+export const useTransportChangeInSelect = (
+    transports: TransportWithAvailablePaymentsAndStoresFragmentApi[] | undefined,
+    lastOrderPickupPlace: ListedStoreFragmentApi | null,
+    changeTransportHandler: ChangeTransportHandler,
+    callbacks: {
+        onTransportReset?: () => Promise<unknown>;
+    } = {},
+) => {
+    const { defaultLocale } = useDomainConfig();
+    const [preSelectedPickupPlace, setPreSelectedPickupPlace] = useState(lastOrderPickupPlace);
+    const [preSelectedTransport, setPreselectedTransport] =
+        useState<TransportWithAvailablePaymentsAndStoresFragmentApi | null>(null);
+    const clearPacketeryPickupPoint = usePersistStore((store) => store.clearPacketeryPickupPoint);
+    const setPacketeryPickupPoint = usePersistStore((store) => store.setPacketeryPickupPoint);
+    const { transport: currentTransport, pickupPlace: currentPickupPlace } = useCurrentCart();
+
+    const changeTransport = async (updatedTransportUuid: string | null): Promise<boolean> => {
+        const updatedTransport = transports?.find((transport) => transport.uuid === updatedTransportUuid);
+
+        if (updatedTransport?.uuid === currentTransport?.uuid) {
+            return false;
+        }
+
+        if (!updatedTransport) {
+            await changeTransportHandler(null, null);
+
+            if (callbacks.onTransportReset) {
+                await callbacks.onTransportReset();
+            }
+
+            return false;
+        }
```

Comment:
I guess it makes more sense to switch those two checks since it's better to check if `updatedTransport` is present and then compare it

```suggestion
        if (!updatedTransport) {
            await changeTransportHandler(null, null);

            if (callbacks.onTransportReset) {
                await callbacks.onTransportReset();
            }

            return false;
        }
        
        if (updatedTransport.uuid === currentTransport?.uuid) {
            return false;
        }
```

Reply:
Could be. I switched them.